python/tank_vendor/shotgun_authentication/session_cache.py

```diff
@@ -0,0 +1,209 @@
+This module will provide basic i/o to read and write session user's credentials
+in the site's cache location.
+"""
+
+import os
+import sys
+import urlparse
+from ConfigParser import SafeConfigParser
+from tank_vendor.shotgun_api3 import Shotgun, AuthenticationFault, ProtocolError
+from tank_vendor.shotgun_api3.lib import httplib2
+from .errors import AuthenticationError
+
+# FIXME: Quick hack to easily disable logging in this module while keeping the
+# code compatible. We have to disable it by default because Maya will print all out
+# debug strings.
+if False:
+    import logging
+    # Configure logging
+    logger = logging.getLogger("sgtk.session_cache")
+    logger.setLevel(logging.DEBUG)
+    logger.addHandler(logging.StreamHandler())
+else:
+    class logger:
+        @staticmethod
+        def debug(*args, **kwargs):
+            pass
+
+        @staticmethod
+        def info(*args, **kwargs):
+            pass
+
+        @staticmethod
+        def warning(*args, **kwargs):
+            pass
+
+        @staticmethod
+        def error(*args, **kwargs):
+            pass
+
+        @staticmethod
+        def exception(*args, **kwargs):
+            pass
+
+
+def _get_local_site_cache_location(base_url):
+    """
+    Returns the location of the site cache root based on a site.
+
+    :param base_url: Site we need to compute the root path for.
+
+    :returns: An absolute path to the site cache root.
+    """
+    if sys.platform == "darwin":
+        root = os.path.expanduser("~/Library/Caches/Shotgun")
+    elif sys.platform == "win32":
+        root = os.path.join(os.environ["APPDATA"], "Shotgun")
+    elif sys.platform.startswith("linux"):
+        root = os.path.expanduser("~/.shotgun")
+
+    # get site only; https://www.foo.com:8080 -> www.foo.com
+    base_url = urlparse.urlparse(base_url)[1].split(":")[0]
+
+    return os.path.join(root, base_url)
+
+
+def _get_session_data_location(base_url):
+    """
+    Returns the location of the session file on disk for a specific site.
+
+    :param base_url: The site we want the login information for.
+
+    :returns: Path to the login information.
+    """
+    return os.path.join(
+        _get_local_site_cache_location(base_url),
+        "authentication",
+        "login.ini"
+    )
+
+
+def delete_session_data(host):
+    """
+    Clears the session cache for the given site.
+
+    :param host: Site to clear the session cache for.
+    """
+    if not host:
+        logger.error("Current host not set, nothing to clear.")
+        return
+    logger.debug("Clearing session cached on disk.")
+    try:
+        info_path = _get_session_data_location(host)
+        if os.path.exists(info_path):
+            logger.debug("Session file found.")
+            os.remove(info_path)
+            logger.debug("Session cleared.")
+        else:
+            logger.debug("Session file not found: %s", info_path)
+    except:
+        logger.exception("Couldn't delete the session cache file!")
```

Comment:
I think this should raise on failure. If you run `delete_session_data` I guess you really have the intent to remove the data from disk. And say for example that someone has made the file readonly - in that case, the method will return and you won't know whether it was successful or not. I think in this particular instance, failing to clear the session data should probably generate an error that bubbles up to the user: you tried to clear all this login info but you failed because the file is readonly. that sort of thing. What do you think?